internal/pkg/chainsync/internal/syncer/syncer.go

```diff
@@ -425,22 +419,28 @@ func (syncer *Syncer) widen(ctx context.Context, ts block.TipSet) (block.TipSet,
 	return wts, nil
 }
 
-// HandleNewTipSet extends the Syncer's chain store with the given tipset if they
-// represent a valid extension. It limits the length of new chains it will
-// attempt to validate and caches invalid blocks it has encountered to
-// help prevent DOS.
-func (syncer *Syncer) HandleNewTipSet(ctx context.Context, ci *block.ChainInfo) (err error) {
+// HandleNewTipSet validates and syncs the chain rooted at the provided tipset
+// to a chain store.  Iff catchup is false then the syncer will set the head.
+func (syncer *Syncer) HandleNewTipSet(ctx context.Context, ci *block.ChainInfo, catchup bool) error {
+	err := syncer.handleNewTipSet(ctx, ci)
+	if err != nil {
+		return err
+	}
+	if catchup {
+		return nil
+	}
+	return syncer.SetStagedHead(ctx)
+}
+
+func (syncer *Syncer) handleNewTipSet(ctx context.Context, ci *block.ChainInfo) (err error) {
```

Comment:
Like that you're not making this method even bigger, but refactoring into nested function calls isn't the best solution for breaking code out. It would be nice if this could get an overall refactor, but it's not necessary for this PR.